공연 상세 조회 컨트롤러 추가

gaya8487 opened the pull request:
공연 상세 조회 컨트롤러를 추가 하였습니다.

gaya8487 committed:
공연 상세 조회 컨트롤러 추가

gaya8487 committed:
불필요한 코드 제거 및 이름 변경

kimjunbo commented on app/src/main/java/com/atticket/reservation/controller/ProductSearchController.java:
```diff
@@ -0,0 +1,53 @@
+import com.atticket.reservation.dto.ProductDetailSearch;
+import com.atticket.reservation.dto.ProductDetail;
+
+@Controller
```
이 부분은 API마다 @ResopnseBody를 붙이기 번거로우니 @RestController를 사용하면 좋을 것 같습니다!

gaya8487 replied:
고쳤습니다!

kimjunbo commented on app/src/main/java/com/atticket/reservation/controller/ProductSearchController.java:
```diff
@@ -0,0 +1,53 @@
+public class ProductSearchController {
+
+	//공연 상세 조회
+	@GetMapping("/products/detailSearch")
```
저번 회의 때 URL 규칙 정한대로 @GetMapping("/products/{id}")로 수정하면 좋을 듯 싶습니다!

gaya8487 replied:
반영 완료했습니다!

f-lab-lyan commented on .editorconfig:
```diff
@@ -0,0 +1,29 @@
+# top-most EditorConfig file
```
이 파일은 어디에서 사용하는 파일인가요?

gaya8487 replied:
.editorconfig 파일을 생성하고 코드 스타일 설정을 적으면,
Editor 관계없이 일정한 코드 스타일을 유지할 수 있다고하여서 설정하여 보았습니다.

f-lab-lyan commented on app/build.gradle:
```diff
@@ -14,6 +14,8 @@ repositories {
 
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.projectlombok:lombok'
+    annotationProcessor 'org.projectlombok:lombok'
```
이것보다는 lombok plugin을 적용하는게 더 깔끔해 보이지 않을까요?

gaya8487 replied:
lombok plugin을 따로 적용하면  spring의 dependency-management 플러그인의 dependency  버전 관리를 받을 수 있나요?

f-lab-lyan commented on app/src/main/java/com/atticket/reservation/controller/ProductSearchController.java:
```diff
@@ -0,0 +1,53 @@
+	//공연 상세 조회
+	@GetMapping("/products/detailSearch")
+	@ResponseBody
+	public ProductDetail productDetail(@ModelAttribute ProductDetailSearch productDetailSearch) {
```
`@ ModelAttribute`이 어노테이션을 쓰신 이유가 있는걸까요?

gaya8487 replied:
조회 시 받아올 파라미터 값이 많을 거라 생각하고 , 
 DTO 로 파라미터 값을 한 번에 가져오기 위해@ModelAttribute를 사용했습니다.

그런데 찾아보니 
multipart/form-data 형태의 데이터를 매핑해주는 @ModelAttribute보다는 
Json데이터를 매핑해주는 @RequestBody를 쓰는 것이 나았을 것 같습니다.

f-lab-lyan commented on app/src/main/java/com/atticket/reservation/controller/ProductSearchController.java:
```diff
@@ -0,0 +1,53 @@
+import com.atticket.reservation.dto.ProductDetailSearch;
+import com.atticket.reservation.dto.ProductDetail;
+
+@Controller
```
`@Controller`와 `@RestController`의 차이에 대해서 생각해 보시면, 좋을 것 같습니다.

f-lab-lyan reviewed (commented):
Dto의 이름이 약간 추상적인 느낌이 있는데요. (이걸 글로 설명하기가 쉽지 않은 것 같은데...) GetProductDetailResponse와 같이 좀 더 구체적인 Dto를 만드는게 더 좋지 않을까 싶은 생각이 듭니다. 어떻게 생각하시나요?

gaya8487 committed:
Merge branch 'main' into YHI_dev

# Conflicts:
#	app/build.gradle

gaya8487 committed:
- 공연 조회 컨트럴로를 임시로 reservation 모듈로 위치 변경
- url규칙을 정함에 따라, 공연 상세 조회 URL을 변경
- DTO 이름을 구체적으로 변경

kimjunbo commented on app/src/main/java/com/atticket/reservation/dto/ProductDetail.java:
```diff
@@ -0,0 +1,84 @@
+public class ProductDetail {
+
+	//공연 이름
+	private String name;
```
DTO에는 한번 set하고 바뀔 일이 없기 때문에 필드에 final을 붙이면 좋을것 같습니다.

gaya8487 replied:
DTO는 데이터 전송을 위한 가변 객체라고 알고 있는데, 한번 set하면 바뀔 일이 없나요?

kimjunbo commented on app/src/main/java/com/atticket/reservation/dto/ProductDetailSearch.java:
```diff
@@ -0,0 +1,21 @@
+import lombok.Getter;
+import lombok.Setter;
+
+@Setter
```
한번 객체가 생성되고 set할 필요가 없으니 @Setter보다는 @AllArgsConstructor가 나을것 같습니다

gaya8487 committed:
공연 좌석 조회 기능을 추가 하였습니다.
- 좌석 조회 기능에 사용할 SeatSearchController를 추가하였습니다.
- 공연 날짜/회차별 좌석 Request/Response DTO를 생성하였습니다.
- DTO에 Request와 Response 패키지를 생성하여 DTO를 이동시켰습니다.
- 공연 상세 조회에서도 새로 생성한 좌석 DTO를 사용하도록 변경하였습니다.

gaya8487 committed:
공연 등록 / 공연 내용 삭제 기능을 추가하였습니다.

gaya8487 committed:
Merge branch 'main' into YHI_dev

# Conflicts:
#	common/response/src/main/java/com/atticket/common/response/ResultResponseDto.java
#	common/response/src/main/java/com/atticket/common/response/request/ProductRegisterRequestDto.java
#	common/response/src/main/java/com/atticket/common/response/request/SeatsSearchRequestDto.java
#	common/response/src/main/java/com/atticket/common/response/response/ProductDetailResponseDto.java
#	common/response/src/main/java/com/atticket/common/response/response/SeatsSearchResponseDto.java

gaya8487 committed:
- main 변경사항 머지
  - reservation 패키지에있던 product controller/dto 위치변경
- @build class 레벨로 변경
- immutable 한 dto을 위해 변수 final 적용
- dto 명명 규칙에 따라 이름 변경

kimjunbo commented on product/src/main/java/com/atticket/product/controller/ProductController.java:
```diff
@@ -1,16 +1,18 @@
 package com.atticket.product.controller;
 
-import static com.atticket.common.response.BaseResponse.ok;
+import static com.atticket.common.response.BaseResponse.*;
```
이 부분 * 사용 되어있는데 IntelliJ 기준 Editor> Code Style > Java에서 `Names count to use static import with *` 옵션값이 1로 되어있어서 이 부분 값을 99정도로 높은 숫자 입력하시면 바뀌실 거에요. 
한번 확인 부탁드려요

kimjunbo commented on product/src/main/java/com/atticket/product/controller/ProductController.java:
```diff
@@ -24,17 +26,79 @@ public BaseResponse<SampleDto> getProducts() {
-			throw new BaseException(BaseStatus.TEST_ERROR);
-		}
-		return ok(SampleDto.builder().content(id).build());
+	//
+	// /**
+	//  * 상품 정보 조회
+	//  */
+	// @GetMapping("/{id}")
+	// public BaseResponse<SampleDto> getTicketInfo(@PathVariable("id") String id) throws Exception {
+	// 	if (id.equals("error")) {
+	// 		throw new Exception("unexpected error our");
+	// 	}
+	// 	if (id.equals("test-error")) {
+	// 		throw new BaseException(BaseStatus.TEST_ERROR);
+	// 	}
+	// 	return ok(SampleDto.builder().content(id).build());
+	// }
```
주석 부분은 지워도 될것 같습니다

kimjunbo reviewed (commented):
잠깐 보았는데 아직 저희가 회의로 의논한 부분은 적용이 안된거 같아 다음 PR때 적용하실 계획이신가요?

gaya8487 committed:
- 의논한 설계변경에 따른 수정
- Dto 이름 수정
- 변수 이름 수정
